public/app/core/logs_model.ts

```diff
@@ -245,23 +249,33 @@ export function getSeriesProperties(
   let visibleRange = absoluteRange;
   let resolutionIntervalMs = intervalMs;
   let bucketSize = Math.max(resolutionIntervalMs * pxPerBar, minimumBucketSize);
+  let timeSpanOfLogs;
+  let timeSpanOfRange;
   // Clamp time range to visible logs otherwise big parts of the graph might look empty
   if (absoluteRange) {
-    const earliest = sortedRows[0].timeEpochMs;
-    const latest = absoluteRange.to;
-    const visibleRangeMs = latest - earliest;
+    const earliestTsLogs = sortedRows[0].timeEpochMs;
+    const latestTsLogs = sortedRows[sortedRows.length - 1].timeEpochMs;
+    const earliestTsAbsolute = absoluteRange.from;
+    const latestTsAbsolute = absoluteRange.to;
+
+    // Calculate timespan for received logs and range. This is used to calculate timespan coverage meta info of received logs.
+    // We are rounding this as some ds are rounding range in their requests which resulted in incorrect coverage (in edge cases).
+    timeSpanOfLogs = Math.ceil((latestTsLogs - earliestTsLogs) / 1000);
+    timeSpanOfRange = Math.ceil((latestTsAbsolute - earliestTsAbsolute) / 1000);
+
+    const visibleRangeMs = latestTsAbsolute - earliestTsLogs;
     if (visibleRangeMs > 0) {
       // Adjust interval bucket size for potentially shorter visible range
       const clampingFactor = visibleRangeMs / (absoluteRange.to - absoluteRange.from);
```

Comment:
Wouldn't it always be `clampingFactor=1` now?

Reply:
Oh it's been updated. Look good :) I think it could be even be written as:

```suggestion
      const clampingFactor = visibleRangeMs / requestedRangeMs;
```